resources/particle_vert.glsl

```diff
@@ -0,0 +1,36 @@
+#version 330 core
+
+const int MAX_ORIENTATIONS = 21;
+
+layout(location = 0) in vec3 vertPos;
+layout(location = 1) in vec3 vertNor;
+layout(location = 2) in vec2 vertTex;
+layout(location = 3) in vec3 particleOffset;
+layout(location = 4) in int orientationID;
+
+uniform mat4 P;
+uniform mat4 M;
+uniform mat4 randomMs[MAX_ORIENTATIONS];
```

Comment:
21x4x4x4 + 21x3x3x4 = over 2mb 
What is the purpose of this? This data is being uploaded per particle per frame - which is a lot. I feel like this could be a lot more efficient.

Reply:
We have two options:
1) for all n particles, upload it's individual M and N matrix, which will be 25 floats. That's 25×N floats.
2) Store all possible M and Ns as a uniform once and an index per particle, which will be at maximum 21×25 floats and n ints. Option2 saves significantly more memory and is what is currently being done. I think.

Reply:
Hmm I don't know which is more efficient or how it plays with instancing. If it works it works. It might be worth asking Ian